src/game/behaviors/hidden_star.inc.c

```diff
@@ -47,12 +47,24 @@ void bhv_hidden_star_trigger_loop(void) {
     }
 }
 
+void bhv_bowser_course_red_coin_star_init(void) {
+    if (o->oBehParams2ndByte != 0) {
+        o->oHiddenStarTriggerTotal = o->oBehParams2ndByte;
+        o->oHiddenStarTriggerCounter = gRedCoinsCollected;
+    }
+    else {
+        s16 numRedCoinsRemaining = count_objects_with_behavior(bhvRedCoin);
+        o->oHiddenStarTriggerTotal = numRedCoinsRemaining + gRedCoinsCollected;
+        o->oHiddenStarTriggerCounter = o->oHiddenStarTriggerTotal - numRedCoinsRemaining;
+    }
+}
```

Comment:
yeah ill make it consistent i guess

Reply:
actually, there's nothing to change. The spawning for the respective stars is different and even though the init doesn't have the star spawn conditions, the loop will instantly set it to action 1 which spawns the star after 3 frames. It already works across multiple areas because in the init it sets its counter and total based on gRedCoinsCollected and the number of reds left in the stage, like the other function.

Reply:
Oh huh I clearly missed those details, although I guess the other function also runs under the same situation, just it won't play a cutscene in that case. Non-critical then, though really the main benefit of killing off the bowser functions is just so all of the red coin stuff is in one place.